src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.MultiSegment.cs

```diff
@@ -2185,35 +2185,42 @@ private ConsumeTokenResult ConsumeNextTokenUntilAfterAllCommentsAreSkippedMultiS
             return ConsumeTokenResult.NotEnoughDataRollBackState;
         }
 
-        private bool ConsumeCommentMultiSegment()
+        private bool SkipOrConsumeCommentMultiSegmentWithRollback()
         {
             long prevTotalConsumed = BytesConsumed;
             SequencePosition start = new SequencePosition(_currentPosition.GetObject(), _currentPosition.GetInteger() + _consumed);
             bool ret = SkipCommentMultiSegment(out int tailBytesToIgnore);
```

Comment:
Nit: can we rename `ret` to something more descriptive of what it actually means?